storage/replica.go

```diff
@@ -1154,6 +1115,20 @@ func (r *Replica) executeBatch(batch engine.Engine, ms *engine.MVCCStats, ba *ro
 	// accumulate updates to it.
 	isTxn := ba.Txn != nil
 
+	// Ensure that timestamps on individual requests are offset by an incremental
+	// amount of logical ticks from the base timestamp of the batch request (if
+	// that is non-zero and the batch is not in a Txn). This has the effect of
+	// allowing self-overlapping commands within the batch, which would otherwise
+	// result in an endless loop of WriteTooOldErrors.
+	// TODO(tschottdorf): discuss self-overlap.
+	// TODO(tschottdorf): provisional feature to get back to passing tests.
+	// Have to discuss how we go about it.
+	delta := 1
```

Comment:
done. It's called `fiddleWithTimestamps` (and I'm hoping it won't be around for much longer).